### PR TITLE
Bump type editor dependency

### PR DESCRIPTION
**Pull request by nathggns**
## 🌟 What is the purpose of this PR?

This bumps the type editor package to make use of the PRs linked below, and makes some fixes to make it the new features work.

This isn't blocked by https://github.com/blockprotocol/blockprotocol/pull/1228 but will work in conjunction with it to address https://github.com/blockprotocol/blockprotocol/pull/1210 which broke the sticky footer. It does this by disabling the horizontal scrolling when in edit mode, as its most useful for read only mode which you're more likely to be when on mobile. Horizontal scrolling will be properly implemented in https://app.asana.com/0/1200211978612931/1204283419240984/f

Context: https://hashintel.slack.com/archives/C02LG39FJAU/p1680018604813949?thread_ts=1680014914.897749&cid=C02LG39FJAU

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1204086167143889/1204280837973751/f) _(internal)_

## 🚫 Blocked by

- [x] https://github.com/hashintel/hash/pull/2258
- [x] https://github.com/hashintel/hash/pull/2255
- [x] https://github.com/hashintel/hash/pull/2271
- [x] https://github.com/hashintel/hash/pull/2293

## 🔍 What does this change?

- As of https://github.com/hashintel/hash/pull/2262 the design system (and therefore type editor) need transpiling by the consumer due to CSS imports, which we now do
- Updates design system and type editor dependencies – see linked tickets for new functionality
- Disabled horizontal scrolling when in edit mode, and added a "min-width: max-content"  around the type editor when in read only mode to make horizontal scrolling work again after https://github.com/blockprotocol/blockprotocol/pull/1228 removes the global one introduced by https://github.com/blockprotocol/blockprotocol/pull/1210
- Increased the z-index of the edit bar to ensure it appears above the sticky insert bar
- Apply the fluid fonts styles from the HASH design system

## 📜 Does this require a change to the docs?

No

## ⚠️ Known issues

The box shadow on tables is hidden on the sides when in read only mode – this is an existing bug

## 🐾 Next steps

- https://app.asana.com/0/1200211978612931/1204283419240984/f

## 🛡 What tests cover this?

None

## ❓ How to test this?

See the linked tickets to check which functionality should work.

### CODE REVIEW

**Comment by nathggns** [comment]
Updated this to reflect we'll not merge it until both PR updates to the type editor are merged

**Review by CiaranMn — APPROVED** [review]

**Pull request merged**
